storage/badger/store.go

```
@@ -21,43 +21,236 @@ package badger
+		uuid := strings.Replace(uuidWithHyphen.String(), "-", "", -1)
+
+		err := w.Checkout(&git.CheckoutOptions{Create: true, Force: true, Branch: plumbing.NewBranchReferenceName(uuid)})
+		if err != nil {
+			return err
+		}
+
+		//we have new branch but we don't need to create a tag, we just need to reset to tag
+		tag := getTag(s.dbGitRepository, context)
+		if tag != nil && !tag.Hash.IsZero() {
+			commit, _ := tag.Commit()
+			w.Reset(&git.ResetOptions{
+				Mode:   git.HardReset,
+				Commit: commit.Hash,
+			})
+		}
+
+	}
+
+	s.db, err = badger.Open(s.badgerOptions)
+	if err != nil {
+		return fmt.Errorf("could not open database: %w", err)
+	}
+
+	return nil
+
+}
+
+func (s *Store) newCommit(message string) {
+	s.unlockGit()
+	defer s.lockGit()
+	s.Sync()
+
+	w, _ := s.dbGitRepository.Worktree()
+
+	w.Add("KEYREGISTRY")
+	w.Add("MANIFEST")
+	w.Add("LOCK")
+
+	err := filepath.Walk(s.path, func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() {
+			return nil
+		}
+		if filepath.Ext(path) == ".vlog" || filepath.Ext(path) == ".sst" {
+			w.Add(path[strings.LastIndex(path, "/")+1:])
+		}
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	w.Commit(message, &git.CommitOptions{
+		Author: &object.Signature{
+			Name:  "Flow Emulator",
+			Email: "emulator@onflow.org",
+			When:  time.Now(),
+		},
+	})
+
+}
+
+func (s *Store) openRepository(directory string) (*git.Repository, error) {
+	dbgit, err := git.PlainOpen(directory)
+	if err == nil {
+		return dbgit, err
+	}
+	if err == git.ErrRepositoryNotExists {
+		result, err := git.PlainInit(directory, false)
+		if err == nil {
+			return result, err
+		}
+		return nil, err
+	}
+	return nil, err
+}
+
 // New returns a new Badger Store.
 func New(opts ...Opt) (*Store, error) {
 	badgerOptions := getBadgerOptions(opts...)
-
+	badgerOptions.BypassLockGuard = true
 	db, err := badger.Open(badgerOptions)
 	if err != nil {
 		return nil, fmt.Errorf("could not open database: %w", err)
 	}
+	db.Sync()
```

Explicity ignore error if you are sure there is no need to handle it